runtime/compiler/control/JitDump.cpp

```diff
@@ -657,6 +658,13 @@ runJitdump(char *label, J9RASdumpContext *context, J9RASdumpAgent *agent)
                if (currentMethodBeingCompiled)
                   oldStartPC = currentMethodBeingCompiled->_oldStartPC;
 
+               // The current thread is a compilation thread which may or may not currently hold VM access. The request
+               // for recompilation to generate the jitdump will be performed synchronously for which the code path we
+               // will take will be the same as if we were an application thread. We will take the synchronous request
+               // path in `compileOnSeparateThread` which ends up releasing VM access prior to waiting on the diagnostic
+               // thread to finish the queued compile. To avoid deadlocks we must acquire VM access here.
+               TR::VMAccessCriticalSection requestSynchronousCompilation(TR_J9VMBase::get(jitConfig, crashedThread));
```

Comment:
It's better that you add the brace brackets to explicitly scope the critical section.

Reply:
Fixed in 9cc7a45